cmd/fossa/cmd/test/e2e_test.go

```diff
@@ -0,0 +1,148 @@
+package test_test
+
+import (
+	"encoding/json"
+	"flag"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/apex/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/cli"
+
+	"github.com/fossas/fossa-cli/api/fossa"
+	"github.com/fossas/fossa-cli/cmd/fossa/cmd/test"
+)
+
+var orgID = rand.Intn(1000)
+
+// taskStatus is a struct that imitates the anonymous struct within fossa.Build
+type taskStatus struct {
+	Status string
+}
+
+var testConfigs = []struct {
+	fetcher string
+	project string
+	locator string
+}{
+	{"custom", "testRun", "custom+" + strconv.Itoa(orgID) + "%2FtestRun$1000"},
+	{"custom", "git@github.com:fossas/fossa-cli.git", "custom+" + strconv.Itoa(orgID) + "%2Fgit@github.com:fossas%2Ffossa-cli.git$1000"},
+	{"custom", "http://github.com/fossas/fossa-cli.git", "custom+" + strconv.Itoa(orgID) + "%2Fgithub.com%2Ffossas%2Ffossa-cli$1000"},
+	{"git", "git@github.com:fossas/fossa-cli.git", "git+github.com%2Ffossas%2Ffossa-cli$1000"},
+	{"git", "http://github.com/fossas/fossa-cli.git", "git+github.com%2Ffossas%2Ffossa-cli$1000"},
+	{"git", "testRun", "git+testRun$1000"},
+}
+
+// This function tests how the cli's test command constructs a locator and sends a request to the desired endpoint.
+// Currently this function implements goroutines and channels to prevent test.Run() from logging fatal and killing
+// the process. In the future, once errors are handled more gracefully the test server can be broken out
+// into its own TestServer package for other commands to access.
```

Comment:
I would add something along the lines of state will be asserted against the TestServer (e.g. what urls were called, etc ...) to avoid needing the channel to leak state